packages/nc-gui/components/smartsheet/toolbar/SearchData.vue

```diff
@@ -55,15 +55,19 @@ function onPressEnter() {
 }
 
 const displayColumnLabel = computed(() => {
-  return columns.value?.find((column) => column.primaryValue)?.label
+  const columnLabel = search.value.field.length === 0?columns.value?.find(column => column.primaryValue)?.label:
+  columns.value?.find(column => column.value === search.value.field)?.label
+  return (columnLabel && columnLabel.length > 10)
+    ? columnLabel.slice(0, 10) + '...'
+    : columnLabel;
```

Comment:
1. You can simply use `truncate` class instead of writing your own logic. See the comment at L98.
2. This looks long, packed and hard to read. You may simplify it like

```ts
const displayColumnLabel = computed(() => {
  if (search.value.field) {
    // use search field label if specified
    return columns.value?.find((column) => column.value === search.value.field)?.label
  }
  // use primary value label by default
  return columns.value?.find((column) => column.primaryValue)?.label
})
```

Reply:
@wingkwong  I've used truncate but not working. Is it me doing anything wrong or any other issue ?

Reply:
I've tested locally and it worked fine. `truncate` is added on L98, not in this code.

Reply:
It's done